### DISCUSSION TITLE
Pimcore Portal Engine for Pimcore X

### DISCUSSION THREAD

**Discussion by archana-multi-concept** [pimcore/pimcore#9614] [Q&A / Support]
We have enterprise subscription. So Trying to install portal engine.
following parameter has been set-up
pimcore_portal_engine.elasticsearch.host: Host of your Elasticsearch cluster
pimcore_portal_engine.elasticsearch.index_prefix: Prefix for all index names created by portal engine

Portal-engine composer and enable command run successfully, but when I run install command, it is giving following error

 ./bin/console pimcore:bundle:install PimcorePortalEngineBundle

15:24:29 WARNING   [portalengine.es] Request Failure: ["method" => "DELETE","uri" => "http://host:9200/indexportaluser","port" => 9200,"headers" => ["Host" => ["host"],"Content-Type" => ["application/json"],"Accept" => ["application/json"],"User-Agent" => ["elasticsearch-php/7.13.1 (Linux 5.4.0-63-generic; PHP 8.0.6)"],"x-elastic-client-meta" => ["es=7.13.1,php=8.0.6,t=7.13.1,a=0,cu=7.68.0"]],"HTTP code" => 401,"duration" => 0.004798,"error" => "{"error":{"root_cause":[{"type":"security_exception","reason":"missing authentication credentials for REST request [/indexportaluser]","header":{"WWW-Authenticate":"Basic realm=\"security\" charset=\"UTF-8\""}}],"type":"security_exception","reason":"missing authentication credentials for REST request [/indexportaluser]","header":{"WWW-Authenticate":"Basic realm=\"security\" charset=\"UTF-8\""}},"status":401}"]
15:24:29 WARNING   [portalengine.es] Response ["{"error":{"root_cause":[{"type":"security_exception","reason":"missing authentication credentials for REST request [/indexportaluser]","header":{"WWW-Authenticate":"Basic realm=\"security\" charset=\"UTF-8\""}}],"type":"security_exception","reason":"missing authentication credentials for REST request [/indexportaluser]","header":{"WWW-Authenticate":"Basic realm=\"security\" charset=\"UTF-8\""}},"status":401}"]

                                                                                                                        
 [ERROR] {"error":{"root_cause":[{"type":"security_exception","reason":"missing authentication credentials for REST     
         request [/indexportaluser]","header":{"WWW-Authenticate":"Basic realm=\"security\"                           
         charset=\"UTF-8\""}}],"type":"security_exception","reason":"missing authentication credentials for REST request
         [/p589796portaluser]","header":{"WWW-Authenticate":"Basic realm=\"security\" charset=\"UTF-8\""}},"status":401}

So the question is, How to setup authentication credentials for the request? and Can I give anything in index paramater.

**Comment by fashxp** [comment]
Hi, 
should be possible with starting with the next version. 
Please contact your Pimcore sales contact person to get access to the portal engine repository for early access to the modification. 

BR
Christian

**Reply by archana-multi-concept** [reply]
Ok, thanks for reply.

**Reply by esorthapa** [reply]
@archana-multi-concept, 
Please suggest me in which yml file we need to setup this for pimcore portal engine:
pimcore_portal_engine.elasticsearch.host: Host of your Elasticsearch cluster
pimcore_portal_engine.elasticsearch.index_prefix: Prefix for all index names created by portal engine

**Reply by fashxp** [reply]
see https://pimcore.com/docs/portal-engine/current/Installation/Installation.html#page_Required-Parameter

**Reply by fashxp** [reply]
in any configuration file that is loaded by symfony ... it is a standard symfony configuration.

**Comment by esorthapa** [comment]
Hi Christian Fasching,
I am trying to implement Enterprise PIMcore in the 6.9.2 version in my
system. I also implement a token with Install Enterprise Extensions Via
Composer as suggested in Certificate.
Now I am trying to implement portal engine with:

composer require pimcore/portal-engine

 Beside that, As in Installation of portal Engine and system
Requirements for 6.9

php=7.4

pimcore=6.9.x

Elasticsearch too

and all the Bundle as required as:

>Enterprise Metadata Bundle

>Statistic Explorer

Direct Edit

Also install using command as suggested in link:
https://pimcore.com/docs/portal-engine/current/Installation/Installation.html#page_Required-Parameter

But, during portal engine installation process,I need to setup
Elasticsearch configure
Furthermore, before the portal engine installation is possible, the
following parameter has to be configured in the system:

    pimcore_portal_engine.elasticsearch.host: Host of your Elasticsearch cluster
    pimcore_portal_engine.elasticsearch.index_prefix: Prefix for all
index names created by portal engine

Please suggest:

>Do we need an AdvancedObjectSearchBundle for elasticsearch?

>Next, I have installed Elasticsearch search on a different host server, so how can I connect with pimcore and Elasticsearch or do we have elasticsearch inbuilt in pimcore?

>How can I configure  and which yml file
  pimcore_portal_engine.elasticsearch.host
  pimcore_portal_engine.elasticsearch.index_prefix

Above are these two methods. I tried to implement it in
app/config/config.yml like this:
......
.....

pimcore:
   advanced_object_search:
    # Prefix for index names
    index_name_prefix: 'advanced_object_search_'

    # List of elasticsearch hosts
    es_hosts:
        - 'localhost:9200'

    pimcore_portal_engine.elasticsearch.hosts:
        - 'localhost:9200'
    pimcore_portal_engine.elasticsearch.index_prefix : 'portal_elastic_search'

......
......

>Finally, I have implemented pimcore Enterprise Subscription but I got this message below attachment.


Please suggest it to me as soon as possible.

**Reply by fashxp** [reply]
please try to configure your messages here properly ... its nearly impossible to get what you are asking for. 

I try to answer what I got: 
- no you don't need advanced object search for running portal engine
- to add needed parameters, for example add following to your `app/config/config.yml`: 
```yml
parameters:
    pimcore_portal_engine.elasticsearch.host: '<YOUR_ES_HOST_NAME>'
    pimcore_portal_engine.elasticsearch.index_prefix: '<SOME_PREFIX_FOR_YOUR_INDEX_NAMES>'
```

you also just could add them into you `parameters.yml` if you have one in place. 

Please bare in mind, that Pimcore 6.9 is EOL and not longer supported. Also 6.9 support by portal engine will not last much longer anymore. If you start a new project, I would highly recommend to start with Pimcore X
